index-handlers/src/main/java/no/sikt/nva/nvi/index/SearchNviCandidatesHandler.java

```diff
@@ -204,9 +155,64 @@ private static String extractQueryParamAssignee(RequestInfo requestInfo) {
     }
 
     @JacocoGenerated
-    private static AuthorizedBackendUriRetriever defaultUriRetriver() {
+    private static AuthorizedBackendUriRetriever defaultUriRetriever() {
         return new AuthorizedBackendUriRetriever(new Environment().readEnv("BACKEND_CLIENT_AUTH_URL"),
                                                  new Environment().readEnv("BACKEND_CLIENT_SECRET_NAME"));
     }
+
+    private CandidateSearchParameters getCandidateSearchParameters(RequestInfo requestInfo)
+        throws UnauthorizedException {
+        boolean isAdmin = userIsNviAdmin(requestInfo);
+        var offset = extractQueryParamOffsetOrDefault(requestInfo);
+        var size = extractQueryParamSizeOrDefault(requestInfo);
+        var filter = extractQueryParamFilterOrDefault(requestInfo);
+        var excludeSubUnits = extractQueryParamExcludeSubUnitsOrDefault(requestInfo);
+        var topLevelOrg = !isAdmin ? requestInfo.getTopLevelOrgCristinId().orElseThrow() : null;
+        var affiliations = getAffiliations(requestInfo, topLevelOrg);
+        var username = requestInfo.getUserName();
+        var searchTerm = extractQueryParamSearchTermOrDefault(requestInfo);
+        var year = extractQueryParamPublicationDateOrDefault(requestInfo, isAdmin);
+        var category = extractQueryParamCategoryOrDefault(requestInfo);
+        var title = extractQueryParamTitle(requestInfo);
+        var contributor = extractQueryParamContributor(requestInfo);
+        var assignee = extractQueryParamAssignee(requestInfo);
+
+        if (!isAdmin) {
+            assertUserIsAllowedToSearchAffiliations(affiliations, topLevelOrg);
+        }
+
+        return new CandidateSearchParameters(searchTerm, affiliations, excludeSubUnits,
+                                             filter, username,
+                                             year, category, title, contributor,
+                                             assignee, topLevelOrg,
+                                             getResultParameters(size, offset));
+    }
+
+    private void assertUserIsAllowedToSearchAffiliations(List<URI> affiliations, URI topLevelOrg)
+        throws UnauthorizedException {
+        var allowed = attempt(() -> this.uriRetriever.getRawContent(topLevelOrg, APPLICATION_JSON)).map(
+                Optional::orElseThrow)
+            .map(str -> createModel(dtoObjectMapper.readTree(str)))
+            .map(model -> model.listObjectsOfProperty(model.createProperty(HAS_PART_PROPERTY)))
+            .map(node -> node.toList().stream().map(RDFNode::toString))
+            .map(s -> Stream.concat(s, Stream.of(topLevelOrg.toString())))
+            .orElseThrow()
+            .collect(Collectors.toSet());
+
+        var illegal = Sets.difference(new HashSet<>(affiliations.stream().map(URI::toString)
```

Comment:
I think because you don't want to actually remove any affiliations from the list of affiliations. These are used later in the code. Therefore, making a copy?
Also, `x.removeAll(y)` returns a boolean
See latest commit